### PR TITLE
checkout-2

### PR DESCRIPTION
**Pull request by puthsaomey98**
Hello bong,

Please help to check my merge request for Checkout Process.

Thanks.

### CODE REVIEW

File: src/main/java/com/udemy/elearning/services/CheckoutCourseService.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    public CheckoutCourseService(CheckoutCourseRepository checkoutCourseRepository) {
+        this.checkoutCourseRepository = checkoutCourseRepository;
+    }
+
+    public CheckoutCourse create(CheckoutCourseRequest checkoutCourseRequest){
+        CheckoutCourse checkoutCourse = new CheckoutCourse();
+        checkoutCourse.setCourseId(checkoutCourse.getCourseId());
+        checkoutCourse.setPrice(checkoutCourse.getPrice());
+        CheckoutCourse checkoutCourseResult = checkoutCourseRepository.save(checkoutCourse);
```
Comment (camboYY) [inline src/main/java/com/udemy/elearning/services/CheckoutCourseService.java:29]: you have save twice  CheckoutCourse checkoutCourseResult = checkoutCourseRepository.save(checkoutCourse);
        return  checkoutCourseRepository.save(checkoutCourse);

File: src/main/java/com/udemy/elearning/services/CheckoutService.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        Checkout checkout1 = checkoutRepository.save(checkout);
+
+
+
+        for (long courseId: checkoutRequest.getCourseId()){
+            CheckoutCourse checkoutCourse = new CheckoutCourse();
+            checkoutCourse.setCourseId(courseId);
+            checkoutCourse.setCheckoutId(courseId);
+            Course course = courseRepository.findById(courseId).orElseThrow(()->new NotFoundException("Course not found"));
+            checkoutCourse.setPrice(course.getPrice());
```
Comment (camboYY) [inline src/main/java/com/udemy/elearning/services/CheckoutService.java:53]: when you have 
 for (long courseId: checkoutRequest.getCourseId()){
            CheckoutCourse checkoutCourse = new CheckoutCourse();
            checkoutCourse.setCourseId(courseId);
            checkoutCourse.setCheckoutId(courseId);
            Course course = courseRepository.findById(courseId).orElseThrow(()->new NotFoundException("Course not found"));
            checkoutCourse.setPrice(course.getPrice());
            CheckoutCourse checkoutCourseResult = checkoutCourseRepository.save(checkoutCourse);
        }


please add  @Transactional annotation on top of Checkout method

File: src/main/java/com/udemy/elearning/configs/SecurityConfiguration.java [hunk 3]
```diff
@@ -35,7 +35,8 @@ public class SecurityConfiguration {
             "api/courseReviews/**",
             "api/users/validate/**",
             "api/users/list",
-            "api/checkout/**"
+            "api/checkout/**",
```
Comment (camboYY) [inline src/main/java/com/udemy/elearning/configs/SecurityConfiguration.java]: do not add "api/checkout/**", into whiltelist , we need to have bearer token

**Review by camboYY — CHANGES_REQUESTED** [review]

File: src/main/java/com/udemy/elearning/services/CheckoutCourseService.java (hunk 1 above)
Comment (puthsaomey98) [reply src/main/java/com/udemy/elearning/services/CheckoutCourseService.java:29]: Yes bong, this mean we can checkout with multiple course.

File: src/main/java/com/udemy/elearning/configs/SecurityConfiguration.java (hunk 3 above)
Comment (puthsaomey98) [reply src/main/java/com/udemy/elearning/configs/SecurityConfiguration.java]: OK bong.

File: src/main/java/com/udemy/elearning/services/CheckoutService.java (hunk 2 above)
Comment (puthsaomey98) [reply src/main/java/com/udemy/elearning/services/CheckoutService.java:53]: What is purpose with @transactional bong?

File: src/main/java/com/udemy/elearning/configs/SecurityConfiguration.java (hunk 3 above)
Comment (puthsaomey98) [reply src/main/java/com/udemy/elearning/configs/SecurityConfiguration.java]: I'm already remove bong.

**Pull request merged**